### PR TITLE
PR GPS reset command

### PR DESCRIPTION
**Pull request by MatejFranceskin**
Add GPS reset command.

This solves open issue from GPSDrivers repo: Add UBX-CFG-RST to force a Cold Start #26
https://github.com/PX4/GpsDrivers/issues/26

There are 3 types of GPS reset:
Hot - In hot start mode, the receiver was powered down only for a short time (4 hours or less), so that its ephemeris is still valid. Since the receiver doesn't need to download ephemeris again, this is the lastest startup method.
Warm - In warm start mode, the receiver has approximate information for time, position, and coarse satellite position data (Almanac). In this mode, after power-up, the receiver normally needs to download ephemeris before it can calculate position and velocity data.
Cold - In cold start mode, the receiver has no information from the last position at startup. Therefore, the receiver must search the full time and frequency space, and all possible satellite numbers. If a satellite signal is found, it is tracked to decode the ephemeris, whereas the other channels continue to search satellites. Once there is a sufficient number of satellites with valid ephemeris, the receiver can calculate position and velocity data.

Currently UBX and SBF drivers support GPS reset, others will return "non implemented" message.

### CODE REVIEW

File: src/drivers/gps/gps.cpp [hunk 1]
```diff
@@ -950,9 +995,41 @@ GPS::publishSatelliteInfo()
 	}
 }
 
-int GPS::custom_command(int argc, char *argv[])
+int
+GPS::custom_command(int argc, char *argv[])
 {
-	return print_usage("unknown command");
+	// Check if the driver is running.
+	if (!is_running() && _object.load() == nullptr) {
```
Comment (bkueng) [inline src/drivers/gps/gps.cpp]:
```suggestion
	if (!is_running()) {
```
is enough.

File: src/drivers/gps/gps.cpp [hunk 2]
```diff
@@ -950,9 +995,41 @@ GPS::publishSatelliteInfo()
...
+		PX4_INFO("not running");
+		return PX4_ERROR;
+	}
+
+	GPS *_instance = get_instance();
+
+	bool res = false;
+
+	if (_instance && argc == 2 && !strcmp(argv[0], "reset")) {
```
Comment (bkueng) [inline src/drivers/gps/gps.cpp]:
```suggestion
	if (argc == 2 && !strcmp(argv[0], "reset")) {
```

File: src/drivers/gps/gps.cpp [hunk 3]
```diff
@@ -997,6 +1078,7 @@ gps start -d /dev/ttyS3 -e /dev/ttyS4
 	PRINT_MODULE_USAGE_PARAM_STRING('p', nullptr, "ubx|mtk|ash|eml", "GPS Protocol (default=auto select)", true);
 
 	PRINT_MODULE_USAGE_DEFAULT_COMMANDS();
+	PRINT_MODULE_USAGE_COMMAND_DESCR("reset", "hot|warm|cold");
```
Comment (bkueng) [inline src/drivers/gps/gps.cpp]:
```suggestion
	PRINT_MODULE_USAGE_COMMAND_DESCR("reset", "Reset GPS device");
	PRINT_MODULE_USAGE_ARG("cold|warm|hot", "Specify reset type", false);
```

**Comment by MatejFranceskin** [comment]
Thanks for the improvements - I commited them.

**Pull request closed without merging**